app/src/androidTest/java/com/android/sample/model/preferences/PreferencesLocalCacheTest.kt

```diff
@@ -0,0 +1,55 @@
+package com.android.sample.model.preferences
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.nullValue
+import org.junit.Before
+import org.junit.Test
+
+class PreferencesLocalCacheTest {
+
+  private lateinit var localCache: PreferencesLocalCache
+  private val preferences = Preferences(unitsSystem = UnitsSystem.METRIC, weight = WeightUnit.KG)
+
+  @Before
+  fun setUp() {
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    localCache = PreferencesLocalCache(context)
+  }
+
+  @Test
+  fun getPreferences_returns_cached_data() = runTest {
+    localCache.savePreferences(preferences)
+
+    val cachedPreferences = localCache.getPreferences().first()
+
+    // Verify the cached preferences match the original data
+    assertThat(cachedPreferences, `is`(preferences))
+  }
+
+  @Test
+  fun clearPreferences_removes_cached_data() = runTest {
+    localCache.savePreferences(preferences)
+
+    localCache.clearPreferences()
+
+    // Verify the cache is cleared
+    val cachedPreferences = localCache.getPreferences().first()
+    assertThat(cachedPreferences, `is`(nullValue()))
+  }
+
+  @Test
+  fun getPreferences_returns_null_when_no_data_cached() = runTest {
+    // Ensure no data is cached
+    localCache.clearPreferences()
+
+    val cachedPreferences = localCache.getPreferences().first()
+
+    // Verify that null is returned when no data is cached
+    assertThat(cachedPreferences, `is`(nullValue()))
```

Comment:
I think this test is very similar to the previous, you could keep only the previous as it is more complete.